### PR TITLE
Use logical AND, OR in conditions

### PR DESCRIPTION
**Pull request by bhaskarmelkani**
## TL;DR;

Use logical AND (&&), OR (||) for conditions.
## Username

@[bhaskarmelkani](https://www.twitter.com/bhaskarmelkani)

Extra

### CODE REVIEW

**Comment by loverajoel** [comment]
@bhaskarmelkani hey use the PR template -> https://github.com/loverajoel/jstips/blob/gh-pages/GIT_TEMPLATE.md

**Comment by bhaskarmelkani** [comment]
@loverajoel : Updated check if needs any modification .

**Comment by xianxingg** [comment]
**Interesting.**
after using coffeescript, i always want to write code in pure javascript like 

``` javascript
  if (val is true or val is false) ...
```

Your tip answer this question in a way.

**But**, it'll be better if, it's not a must that isTrue() comes along with &&, isFalse() comes along with ||.
I don't want to have a second though for using `&&`/`||` whenever i test a variable is true or false (Though it may be natural after several times).

**I** thought about this:

``` javascript
  function isTrue(v) { return v; }
  function isFalse(v) { return !v; }
```

but it means you must wrap your target value, which is a little ugly and less elegant.

**Comment by bhaskarmelkani** [comment]
@loverajoel Can I add a use-case for &&.

The logical AND could be used to avoid exceptions when using properties of undefined.
Example:-

```
var dog = { 
  bark: function(){
     console.log('Woof Woof');
   }
};

// Calling dog.bark();
dog.bark(); // Woof Woof.

//But if dog is not defined, dog.bark() will raise an error "Cannot read property 'bark' of undefined."
// To prevent this, we can you &&.

dog&&dog.bark();   // This will only call dog.bark(), if dog is defined.

```

**Comment by xianxingg** [comment]
`dog && dog.bark && dog.bark();`

**Comment by yuvarajvemula** [comment]
Nice tip :+1: @bhaskarmelkan

**Comment by yuvarajvemula** [comment]
We can even add this && use for best security practices of JavaScript .. How about this ya 
@bhaskarmelkan

**Comment by loverajoel** [comment]
@bhaskarmelkani thanks for your tip!

**Comment by bhaskarmelkani** [comment]
@loverajoel your welcome :)

File: _posts/en/2016-xx-xx-short-circiut-evaluation-in-js.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  isFalse();    // Test is false.
+}
+
+( test || isFalse());  // Test is false.
+```
+The logical OR could also be used to set a default value for function argument.
+```
+function theSameOldFoo(name){ 
+    name = name || 'Bar' ;
+    console.log('My best friend's name is ' + name);
```
Comment (bhaskarmelkani) [inline _posts/en/2016-xx-xx-short-circiut-evaluation-in-js.md:51]: console.log("My best friend's name is " + name);

**Pull request merged**